server/src/test/java/org/elasticsearch/indices/settings/UpdateSettingsIT.java

```diff
@@ -647,4 +649,18 @@ public void testNumberOfReplicasSettingsVersion() {
         assertThat(newSettingsVersion, equalTo(1 + settingsVersion));
     }
 
+    public void testUpdateSettingsNormalization() {
+        createIndex("test");
+        ensureGreen("test");
+
+        for (String setting : Arrays.asList(SETTING_READ_ONLY, SETTING_BLOCKS_METADATA, SETTING_READ_ONLY_ALLOW_DELETE)) {
+            String settingWithoutPrefix = setting.substring(INDEX_SETTING_PREFIX.length());
+
+            Settings settings = Settings.builder().put(settingWithoutPrefix, true).build();
+            assertAcked(client().admin().indices().prepareUpdateSettings("test").setSettings(settings));
```

Comment:
I believe this test ought to check that the value of the setting actually changes, for example by checking the result of `client().admin().indices().prepareGetSettings("test").get();`.